vochain/transaction/transaction.go

```diff
@@ -434,6 +439,41 @@ func (t *TransactionHandler) CheckTx(vtx *vochaintx.Tx, forCommit bool) (*Transa
 			return response, nil
 		}
 
+	case *models.Tx_SetSik:
+		err := t.SetSikTxCheck(vtx)
```

Comment:
this method might return the address and the newSik, so you don't need to recalculate/refetch it again.